frontend/src/services/ApiService.js

```diff
@@ -1,146 +1,152 @@
-        }
-        else{
-            this.basePath = "http://localhost:8080/api/";
-        }
+class ApiService {
+  static get authPath() {
+    return 'auth/login';
+  }
+
+  constructor() {
+    // Save the JWT token here when acquired
+    this.jwtToken = window.sessionStorage.jwtToken;
+
+    // API basepath, will be processed by webpack accordingly
+    if (process.env.NODE_ENV === 'production') {
+      // this.basePath = "https://bmn.a-eskwadraat.nl/api/";
+      this.basePath = 'http://localhost:8080/api/';
+    } else {
+      this.basePath = 'http://localhost:9000/api/';
     }
-    /**
+  }
+
+  /**
      * Returns the appropriate request method for the given CRUD method
      * @param {int} crudMethod One of the Methods enumeration
      * @returns {String} String of request method to use. Invalid method defaults to GET.
      */
-    static getRequestMethod(crudMethod){
-        switch(crudMethod){
-            case Methods.CREATE:
-            case Methods.UPDATE:
-                return "POST";
-            case Methods.READ:
-                return "GET";
-            case Methods.DELETE:
-                return "DELETE";
-            default:
-                return "GET";
-        }
+  static getRequestMethod(crudMethod) {
+    switch (crudMethod) {
+      case Methods.CREATE:
+      case Methods.UPDATE:
+        return 'POST';
+      case Methods.READ:
+        return 'GET';
+      case Methods.DELETE:
+        return 'DELETE';
+      default:
+        return 'GET';
     }
-    
-    /**
+  }
+
+  /**
      * Perform the special authentication request, which results in a JWT token to be used
-     * in subsequent authorization calls. 
+     * in subsequent authorization calls.
      * TODO: does JWT expire?
      * @param {string} user
      * @param {string} pass
      * @returns {Promise} Promise that is resolved with a JS object with token and username, or rejected
      *  with an ApiError class instance.
      */
-    authRequest(user, pass){
-        //Setup request with parameters
-        let req = new JSONRequest({username: user, password: pass});
-        
-        return req.send("POST", this.basePath + ApiService.authPath)
-        .then(data=>{
-            try{
-                this.jwtToken = data.token;
-            }
-            catch(e){
-                return Promise.reject(new ApiError("Could not decode JSON response for authentication",-1));
-            }
-           
-           //Return meaningful data to 
-           return Promise.resolve({token:data.token, username:user});
-        });
-    }
-    
-    /**
+  authRequest(user, pass) {
+    // Setup request with parameters
+    const req = new JSONRequest({ username: user, password: pass });
+
+    return req.send('POST', this.basePath + ApiService.authPath)
+      .then((data) => {
+        try {
+          this.jwtToken = data.token;
+          window.sessionStorage.jwtToken = data.token;
```

Comment:
Bij mij werkte dit niet? Wel via "window.sessionStorage.setItem('jwtToken',data.token)". Misschien een browser discrepantie? [Doc](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage)

Reply:
Gefixt geloof ik. Kan je kijken of het nu ook echt werkt bij jou?